public/app/features/dashboard-scene/scene/RowRepeaterBehavior.ts

```diff
@@ -90,6 +107,15 @@ export class RowRepeaterBehavior extends SceneObjectBase<RowRepeaterBehaviorStat
     };
   }
 
+  private _getRows(layout: SceneGridLayout, rowToRepeat: SceneGridRow): SceneGridRow[] {
+    const clonedRows: SceneGridRow[] = layout.state.children.filter(
+      (child): child is SceneGridRow =>
+        !!(child instanceof SceneGridRow && child.state.key?.includes(`${rowToRepeat.state.key}-clone-`))
+    );
+
+    return [rowToRepeat, ...clonedRows];
+  }
```

Comment:
maybe the behavior can keep a reference to a list of all the rows it created last performRepeat (including the first template row), that way can just use that instead of looking it up like this